backend/app-kubernetes-service/src/main/kotlin/services/UtilizationService.kt

```diff
@@ -0,0 +1,141 @@
+import dk.sdu.cloud.app.orchestrator.api.JobUtilization
+import dk.sdu.cloud.service.k8.*
+
+/**
+ * A service responsible for fetching utilization information
+ */
+class UtilizationService(
+    private val k8: K8Dependencies
+) {
+    suspend fun allocatable(): CpuAndMemory {
+        val namespace = kotlin.runCatching {
+            k8.client.getResource<Namespace>(
+                KubernetesResources.namespaces.withName("app-kubernetes")
+            )
+        }.getOrThrow()
+
+        val computeAnnotation = namespace.metadata?.annotations?.get("scheduler.alpha.kubernetes.io/node-selector").toString()
+        val nodes = if (computeAnnotation == "ucloud-compute=true") {
+            kotlin.runCatching {
+                k8.client.listResources<Node>(
+                    KubernetesResources.node.withNamespace(NAMESPACE_ANY)
+                ).items.filter { node ->
+                    node.metadata?.labels?.get("ucloud-compute") == "true"
+                }
+            }
+        } else {
+            kotlin.runCatching {
+                k8.client.listResources(
+                    KubernetesResources.node.withNamespace(NAMESPACE_ANY)
+                )
+            }
+        }
+
+        val nodeAllocatableCpu = nodes.getOrThrow().sumOf { node ->
+            node.status?.allocatable?.cpu?.toDouble() ?: 0.0
+        }
+
+        val nodeAllocatableMemory = nodes.getOrThrow().sumOf { node ->
+            memoryStringToBytes(node.status?.allocatable?.memory)
+        }
+
+        return CpuAndMemory(nodeAllocatableCpu, nodeAllocatableMemory)
+    }
+
+    suspend fun used(): CpuAndMemory {
+        val jobs = runCatching {
+            k8.client.listResources<VolcanoJob>(
+                KubernetesResources.volcanoJob.withNamespace("app-kubernetes")
+            )
+        }.getOrThrow()
+
+        val cpuUsage = jobs.sumOf { job ->
+            job.spec?.tasks?.sumOf { task ->
+                task.template?.spec?.containers?.sumOf { container ->
+                    cpuStringToCores(container.resources?.limits?.get("cpu")?.toString())
+                } ?: 0.0
+            } ?: 0.0
+        }
+
+        val memoryUsage = jobs.sumOf { job ->
+            job.spec?.tasks?.sumOf { task ->
+                task.template?.spec?.containers?.sumOf { container ->
+                    memoryStringToBytes(container.resources?.limits?.get("memory")?.toString())
+                } ?: 0
+            } ?: 0
+        }
+
+        return CpuAndMemory(cpuUsage, memoryUsage)
+    }
+
+    suspend fun jobs(): JobUtilization {
+        val jobs = kotlin.runCatching {
+            k8.client.listResources<VolcanoJob>(
+                KubernetesResources.volcanoJob.withNamespace("app-kubernetes")
+            )
+        }.getOrThrow()
+
+        val running = jobs.filter { job ->
+            job.status?.running == 1
+        }.length
+
+        val pending = jobs.filter { job ->
+            job.status?.pending == 1
+        }.length
```

Comment:
`running` and `pending` refers to how many replicas are currently running. This code wouldn't work if a job was scheduled with multiple replicas.